### PR TITLE
Add vrf support for bridges

### PR DESCRIPTION
**Pull request by MrSuicideParrot**
This pull request adds support for adding a bridge created by netavark to a VRF (#729) .

### CODE REVIEW

**Comment by baude** [comment]
first glance, LGTM, thanks @MrSuicideParrot

File: src/network/bridge.rs [hunk 1]
```diff
@@ -494,6 +498,17 @@ fn create_interfaces(
                     InfoKind::Bridge,
                 );
                 create_link_opts.mtu = data.mtu;
+
+                if data.vrf.is_some() {
+                    let vrf = match host.get_link(netlink::LinkID::Name(
+                        data.vrf.as_ref().unwrap().to_string(),
```
Comment (flouthoc) [inline src/network/bridge.rs:504]: should this be handled `unwrap().to_string()` ? What would happen here is `unwrap()` fails ? There is no wrapping of error here.

File: src/network/netlink.rs [hunk 2]
```diff
@@ -37,6 +37,7 @@ pub struct CreateLinkOptions<'fd> {
     pub link: u32,
     pub mac: Vec<u8>,
     pub netns: Option<BorrowedFd<'fd>>,
+    pub master: Option<u32>,
```
Comment (Luap99) [inline src/network/netlink.rs:40]: this option already exists as `primary_index` so do not add this again

File: src/network/bridge.rs (hunk 1 above)
Comment (Luap99) [reply src/network/bridge.rs:504]: The best way to write this is `if let Some(vrf) = data.vrf {` then in the if block you can directly access the vrf var.

File: test/600-bridge-vrf.bats [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    run_in_host_netns ip link set dev test-vrf up
+
+    run_netavark --file ${TESTSDIR}/testfiles/simplebridge-vrf.json setup $(get_container_netns_path)
+    result="$output"
+    assert_json "$result" 'has("podman")' == "true" "object key exists"
+
+    mac=$(jq -r '.podman.interfaces.eth0.mac_address' <<<"$result")
+    # check that interface exists
+    run_in_container_netns ip -j --details link show eth0
+
```
Comment (Luap99) [inline test/600-bridge-vrf.bats:19]: no need to check this

File: test/600-bridge-vrf.bats [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+@test vrf - simple bridge {
+    run_netavark --file ${TESTSDIR}/testfiles/simplebridge.json setup $(get_container_netns_path)
+    run_in_host_netns ip -j --details link show podman0
+    result="$output"
+    assert_json "$result" ".[].linkinfo.info_slave_kind" "==" "null" "VRF is not set"
+}
+
+@test vrf - non existent vrf {
+    expected_rc=1 run_netavark --file ${TESTSDIR}/testfiles/simplebridge-vrf.json setup $(get_container_netns_path)
```
Comment (Luap99) [inline test/600-bridge-vrf.bats:34]: IF you check for errors please also check for the errors message

File: test/600-bridge-vrf.bats [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    assert_json "$result" 'has("podman")' == "true" "object key exists"
+
+    mac=$(jq -r '.podman.interfaces.eth0.mac_address' <<<"$result")
+    # check that interface exists
+    run_in_container_netns ip -j --details link show eth0
+
+    # check if vrf exists
+    run_in_host_netns ip -j --details link show podman0
+    result="$output"
+    assert_json "$result" ".[].linkinfo.info_slave_kind" "==" "vrf" "Bridge has a vrf set"
```
Comment (Luap99) [inline test/600-bridge-vrf.bats:23]: This should check the the vrf is in fact `test-vrf`

**Review by Luap99 — CHANGES_REQUESTED** [review]

File: src/network/netlink.rs (hunk 2 above)
Comment (MrSuicideParrot) [reply src/network/netlink.rs]: I removed all the changes that I made to netlink.rs  and used instead  the `primary_index`.

File: test/600-bridge-vrf.bats (hunk 4 above)
Comment (MrSuicideParrot) [reply test/600-bridge-vrf.bats:29]: I added a check to verify if we receive the correct error message.

File: src/network/bridge.rs (hunk 1 above)
Comment (MrSuicideParrot) [reply src/network/bridge.rs]: Thank you very much for your input, I don't have much experience  with rust. I rewrote this part in my latest commit, could you check if it is better?

**Review by flouthoc — APPROVED** [review]
Thanks :) PR LGTM

**Review by Luap99 — APPROVED** [review]
LGTM

**Comment by Luap99** [comment]
@MrSuicideParrot Thanks for the PR, FYI in order to add this option with `podman network create --opt` this needs to be wired into c/common https://github.com/containers/common/blob/e18cda8d7750746031de3d3d9305df059ac5f0ae/libnetwork/netavark/config.go#L174
And then a docs PR to podman to document this new option.

Let me know if you want to take care of that otherwise I will do it later.

**Pull request merged**